configuration_go/abstr/kubernetes/thanos/ruler/syncer.go

```diff
@@ -5,29 +5,52 @@ import (
 	"path/filepath"
 
 	"github.com/observatorium/observatorium/configuration_go/kubegen/cmdopt"
+	"github.com/observatorium/observatorium/configuration_go/kubegen/containeropts"
 	kghelpers "github.com/observatorium/observatorium/configuration_go/kubegen/helpers"
 	"github.com/observatorium/observatorium/configuration_go/kubegen/workload"
 	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"gopkg.in/yaml.v2"
 	corev1 "k8s.io/api/core/v1"
 )
 
 const (
 	defaultSyncerInternalPort = 8083
 )
 
+type TenantsConfig struct {
+	Tenants []TenantConfig `yaml:"tenants"`
+}
+
+type TenantConfig struct {
+	ID string `yaml:"id"`
+}
+
+func NewTenantsConfigFile(value *TenantsConfig) *containeropts.ConfigResourceAsFile {
+	ret := containeropts.NewConfigResourceAsFile("/etc/thanos-rules-syncer/tenants", "config.yaml", "objstore", "observatorium-rules-syncer-tenants")
+	if value != nil {
+		valueBytes, err := yaml.Marshal(value)
+		if err != nil {
+			panic(err)
```

Comment:
is panicking here the best solution?

Reply:
It is for manifests file generation. So I assume that you want to know right away if something is wrong. 
You are thinking of another approach?

Reply:
no thats all good